share/completions/sops.fish

```diff
@@ -0,0 +1,147 @@
+# Completion for SOPS (Secrets OPerationS)
+
+function __fish_sops_commands --description "Test if argument(s) match a sops command"
+  if not set -q argv[1]
+    return 1
+  end
+  set -l cmd (commandline -cp)
+  if set -q argv[2]
+    if string match -qr -- "sops\s+$argv[1]\s+$argv[2]\s+\S*" $cmd
```

Comment:
Can you try to replace uses of this function with `__fish_seen_subcommand_from`?

I believe if there are multiple levels of subcommands, it's okay to use `__fish_seen_subcommand_from subcmd && __fish_seen_subcommand_from subsubcmd` (not always fully correct but good enough)

Reply:
Do you mean deleting this whole `__fish_sops_commands` function and use `__fish_seen_subcommand_from` instead?

Reply:
> Do you mean deleting this whole `__fish_sops_commands` function and use `__fish_seen_subcommand_from` instead? 

yep

Reply:
even better, replace it with a function that uses argparse to decide which subcommand is active (similar to `__fish_sudo_print_remaining_args`)

Reply:
The reason why I didn't use `__fish_seen_subcommand_from subcmd1 && __fish_seen_subcommand_from subcmd2` for  multiple levels subcommands is because it accepts anything as long as there are the word `subcmd1` and `subcmd2` ignoring the placement. For example, if I want `group` as `subcmd1` and `add` as `subcmd2` then even if I write it as `sops blablabla add groups` it will still return true. While the function that I'm using it right now (I took the code from `__btrfs_commands` and `__btrfs_command_groups`) will return false unless the command is `sops groups add` and nothing else.

Reply:
I might go with the even better "similar to `__fish_sudo_print_remaining_args`" approach, but it might need some time because I'm still figuring out how does `argparse` work. T.T

Reply:
Ok makes sense. I think copying `__fish_sudo_print_remaining_args` is the way to go. We should probably add more examples to the argparse docs.

Reply:
I have modified the way it works, please have another look @krobelus. I don't know if the code is good as I'm not experienced enough for this (not even sure what I did lol), please correct my code if needed. :)